src/vendor/azure-uamqp-c/src/link.c

```diff
@@ -1575,6 +1570,12 @@ void link_dowork(LINK_HANDLE link)
     {
         tickcounter_ms_t current_tick;
 
+        if (link->current_link_credit == 0)
```

Comment:
I'm not sure about this one.
From the code I think what you proposed to do is by calling this way to get the sender/receiver start working:
```c
while(work_flag) {
    connection_dowork(conn);
    link_dowork(link);
}
```

the responsibility of `send_flow` is shipped from callback to an active way.

I'm wondeing about the case below:

1. The current link credit is 5
2. connection_dowork receives 10 Transfer frames
3. it will call the callback function `link_frame_received`
4. the link_instance->current_link_credit becomes < 0
5. link_dowork is called

Reply:
Yeah - tbh I don't know about this either.... And I need to spend more time testing it.
At the very least I can make this a `<= 0` check in the meantime, in case we ever hang because it somehow went negative...